client/src/pages/Edit.js

```diff
@@ -0,0 +1,103 @@
+import React, { useState, useEffect } from "react";
+import Button from "@mui/material/Button";
+import styled from "styled-components";
+import { useNavigate } from "react-router-dom";
+import axios from "axios";
+
+function Edit() {
+  const navigate = useNavigate();
+  const [editFood, setEditFood] = useState({
+    foodName: "",
+    foodAuthor: "",
+    img: "",
+  });
+
+  useEffect(() => {
+    const editFoodId = async () => {
+      const reqdata = await fetch("http://localhost:5000/food");
+      const response = reqdata.json();
```

Comment:
Since I made you wait so long for this review I'm going to put some extra focus on these two lines of code since there's a lot going on here and it's an excellent opportunity to break down each step.

The `fetch()` function returns a promise that resolves to a [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) object.  Because of this, I typically like to name the variable `response`.  I do this since it's not technically the data _yet_.  That means we'd need to name the next variable something else since it's using the word we want.  I like to name this one `result`; in my head, it's the "result" of the `.json()` transformation.   

Also, the `.json()` method of a response object is an asynchronous call, so you'll need to `await` the result.

``` javascript
const response = await fetch("http://localhost:5000/food");
const result = await response.json();
```

This approach works fine when the server sends back a `200` response.  If, on the other hand, the server returns a non-`200` response, this code will fail.  Specifically, it will throw an error when attempting to parse the non-200 response as JSON because it won't _be_ JSON, it will be plain text; such as `500 Application Error` or `401 Unauthorized`.  At this point, you have a few ways to approach it, but I prefer to check the response object's [`ok`](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok) property to determine if the response was successful.

``` javascript
const response = await fetch("http://localhost:5000/food");
let result = null;
if (response.ok) {
  result = await response.json();
} else {
  result = await response.text();
}
return result;
```

Now we're looking much better.  No errors on our watch!  💪 

Although much improved, there are still a few rough edges with this implementation.  For example, how will the calling code know when an error occurred at the server and therefore the return value is plain text, not an object?  It's possible to handle this at the call site, but that's a lot of code to _remember_ to write each time.  We're bound to skip it or forget it.  

What if we always sent back an object with a predictable, structured set of fields?  This way we _always_ know it's an object and we _always_ know that if `data` is populated, `error` isn't; the inverse is also true.

```suggestion
      const response = await fetch("http://localhost:5000/food");
      const result = {
        data: null,
        error: null,
      };
      if (response.ok) {
        result.data = await response.json();
      } else {
        result.error = await response.text();
      }
      return result;
```

There's one more step beyond this one we could take if you wanted to turn this whole thing into a robust tool for your own data fetching needs.  Maybe we talk through it on a future session, or I do another write up like this once you've done this implementation a few times to get the hang of it?